src/service/Review.ts

```diff
@@ -178,167 +178,128 @@ const add = async ({pullRequestLink, slackChannelId, slackMsgId, slackPermalink,
         codeReview,
     };
 }
-const approve = async ({pullRequestLink, reactionUserId}: ReactionData) => {
-    const prisma = new PrismaClient();
-    const codeReview = await prisma.codeReview.findFirst({
-        where: {
-            pullRequestLink,
-        }
-    })
+const approve = async (codeReview: CodeReview & {user: User, reviewers: CodeReviewRelation[]}, slackUserId: string) => {
 
-    if (!codeReview) {
-        return {
-            message: 'Cannot locate existing code review request data.',
-        };
-    }
-
-    const user = await setCodeReviewerStatus(codeReview, reactionUserId, 'approved');
+    const requestSlackUserId = codeReview.user.slackUserId ?? '';
```

Comment:
Can this be an empty string or will that throw errors later?

Reply:
Because the slackUserid is a nullable value, it resolve to string | null as data type by definition from prisma model.  

As long this value is a string, we should be good, slack api won't throw an error in the message text.  We use this value in line 199 which require it to be a string; otherwise the ts compiler will throw error due to strict typing.

This value should not be empty unless the PR is entered via WebUI where we don't know info about the slack account.  We might need to deal with that when we start working on the Web UI login portion to make sure the user have a slack account before allow them to login.

Alternative is to fix the User model to make sure slackUserId is not null: https://github.com/penske-media-corp/code-review-bot/blob/main/prisma/schema.prisma#L54

Reply:
Fixed in commit [c64741f](https://github.com/penske-media-corp/code-review-bot/pull/4/commits/c64741feace5cef14a5bc7a0208694dbba5feb5e)